recipes/napari-hdf5-labels-io/meta.yaml

```
@@ -0,0 +1,48 @@
+{% set name = "napari-hdf5-labels-io" %}
+{% set version = "0.3.dev16" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/napari-hdf5-labels-io-{{ version }}.tar.gz
+  sha256: a16a66b3480b374403f6dd64d03de5cf3624d3072783f330d7d2baf7f613dac6
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python >=3.6, <3.9
+    - setuptools-scm
+  run:
+    - h5py ==2.10.0
```

Is this the only known version to work @pranjaldhole? We usually prefer more relaxed constraints to make it easier for the conda solver.

---

@dlesmesl Do you remember why we have explicit dependency on the `h5py==2.10.0`? I quickly tested if we can work with the newer h5py version but I got some errors related to comparators. Is it going to be easy to upgrade `h5py` dependency to a recent version?

---

@pranjaldhole As long as I remember it was due to version compatibility with `TensorFlow`. At the time we developed the plugins `TensorFlow` was still not supported in `Python 3.9` and the latest version of `h5py ` was developed to work there.

The main issue I remember we had also was that accessing some data in the .h5 file is different from `2.10.0` and the newer versions. On the other hand, we were also having some issues with `pyilastik` which also depends on an old version of `h5py` but this should not be a problem anymore since we were planning to discontinue it.

Currently, I am not completely sure if YAPiC is now supported in `Python  3.9`. If this is the case, I would say there should not be so difficult to upgrade the package.

---

I have created an issue to make the `h5py` dependency more relaxed https://github.com/yapic/napari-hdf5-labels-io/issues/15 . We will deal with it as soon as we can.

---

Thanks! If that's ok then we'll wait a bit more in case you manage to fix that while we migrate the rest of the ecosystem. Please do ping us when a new release is ready or if you'd rather have it released asap.

---

I think we can make the release of the conda package with a relaxed constrained @jaimergp : )

---

```suggestion
    - h5py >=2.10.0
```

Let's try like this for now